Mathlib/Tactic/Linter/Header.lean

```diff
@@ -0,0 +1,287 @@
+    -- Validate the first copyright line.
+    let copStart := "Copyright (c) 20"
+    let copStop := ". All rights reserved."
+    if !copyrightAuthor.startsWith copStart then
+      output := output.push
+        (toSyntax copyright (copyrightAuthor.take copStart.length),
+         s!"Copyright line should start with 'Copyright (c) YYYY'")
+    if !copyrightAuthor.endsWith copStop then
+      output := output.push
+        (toSyntax copyright (copyrightAuthor.takeRight copStop.length),
+         s!"Copyright line should end with '. All rights reserved.'")
+    -- Validate the authors line.
+    let authorsLine := "\n".intercalate authorsLines.dropLast
+    let authorsStart := (("\n".intercalate [openComment, copyrightAuthor, license, ""])).endPos
+    output := output.append (authorsLineChecks authorsLine authorsStart)
+    let expectedLicense := "Released under Apache 2.0 license as described in the file LICENSE."
+    if license != expectedLicense then
+      output := output.push (toSyntax copyright license,
+        s!"Second copyright line should be \"{expectedLicense}\"")
+  | _ =>
+    output := output.push (toSyntax copyright "-/", s!"Copyright too short!")
+  return output
+
+/-- Check the `Syntax` `imports` for broad imports: either `Mathlib.Tactic` or any import
+starting with `Lake`. -/
+def broadImportsCheck (imports : Array Syntax)  : Array (Syntax × String) := Id.run do
+  let mut output := #[]
+  for i in imports do
+    match i.getId with
+    | `Mathlib.Tactic =>
+      output := output.push (i, s!"Files in mathlib cannot import the whole tactic folder.")
+    | modName =>
+      if modName.getRoot == `Lake then
+      output := output.push (i,
+        s!"In the past, importing 'Lake' in mathlib has led to dramatic slow-downs of the linter \
+          (see e.g. mathlib4#13779). Please consider carefully if this import is useful and \
+          make sure to benchmark it. If this is fine, feel free to allow this linter.")
+  return output
+
+/--
+The "header" style linter checks that a file starts with
+```
+/-
+Copyright ...
+Apache ...
+Authors ...
+-/
+
+import*
+module doc-string*
+rest
+```
+It emits a warning if
+* the copyright statement is malformed;
+* `Mathlib.Tactic` is imported;
+* any import in `Lake` is present;
+* the first non-`import` command is not a module doc-string.
+
+The linter allows `import`-only files and does not require a copyright statement in `Mathlib.Init`.
+-/
+register_option linter.style.header : Bool := {
+  defValue := false
+  descr := "enable the header style linter"
+}
+
+namespace Style.header
+
+@[inherit_doc Mathlib.Linter.linter.style.header]
+def headerLinter : Linter where run := withSetOptionIn fun stx ↦ do
+  unless Linter.getLinterValue linter.style.header (← getOptions) do
+    return
+  if (← get).messages.hasErrors then
+    return
+  let mainModule ← getMainModule
+  if mainModule == `Mathlib then return
```

Comment:
Note to self: is this branch needed? Are import-only files excluded anywhere? Continue reviewing from here!

Reply:
I presume the most likely explanation is copy-pasting from the long-file linter. Let's remove this branch.
```suggestion